api_app/api/routes/workspaces.py

```diff
@@ -147,6 +164,16 @@ async def delete_workspace(response: Response, workspace=Depends(get_workspace_b
     return OperationInResponse(operation=operation)
 
 
+@workspaces_core_router.post("/workspaces/{workspace_id}/invoke-action", status_code=status.HTTP_202_ACCEPTED, response_model=OperationInResponse, name=strings.API_INVOKE_ACTION_ON_WORKSPACE)
+async def invoke_action_on_workspace(response: Response, action: str, workspace=Depends(get_workspace_by_id_from_path), resource_template_repo=Depends(get_repository(ResourceTemplateRepository)), operations_repo=Depends(get_repository(OperationRepository)), user=Depends(get_current_workspace_owner_or_researcher_user)) -> OperationInResponse:
```

Comment:
for @marrobi - for `patch` and `delete` ops we require workspace admin - do we want the same here for invoke action?

Reply:
Correct, a researcher should not be able to carry out actions on a workspace. 

The workspace owner is the appropriate role to do this.

Reply:
Amended